packages/tobot_core/src/scene.cpp

```diff
@@ -3,40 +3,77 @@
 using namespace Tobot::Core;
 
 Scene::Scene(const char * id) : m_Id(id) {
+    Layer * baseLayer = new Layer(this->m_BaseLayerId, 0);
+    this->m_Layers[this->m_BaseLayerId] = baseLayer;
 }
 
 void Scene::add(GameEntity * gameEntity) {
-    this->m_GameEntities[gameEntity->getId()] = gameEntity;
+    this->m_Layers.at(this->m_BaseLayerId)->entities[gameEntity->getId()] = gameEntity;
+}
+
+void Scene::add(const char * layerId, GameEntity * gameEntity) {
+    this->m_Layers.at(this->m_BaseLayerId)->entities.insert(std::make_pair(gameEntity->getId(), gameEntity));
+}
+
+void Scene::addLayer(Layer * layer) {
+    this->m_Layers[layer->id] = layer;
 }
 
 void Scene::prepareTextures(SDL_Renderer * renderer) {
-    for (auto object : this->m_GameEntities) {
-        object.second->initializeTexture(renderer);
+    for (auto layer : this->m_Layers) {
+        for (auto entity : layer.second->entities) {
+            entity.second->initializeTexture(renderer);
+        }
     }
 }
 
 void Scene::update() {
-    for (auto object : this->m_GameEntities) {
-        object.second->update();
+    std::vector<Layer *> layers;
+    if (this->m_Layers.size() > 0) {
+        std::vector<Layer *> sortedLayers;
+        for (auto & pair : this->m_Layers) {
+            sortedLayers.push_back(pair.second);
+        }
+        std::sort(sortedLayers.begin(), sortedLayers.end(), [](Layer * a, Layer * b) { return a->order < b->order; });
+        layers = sortedLayers;
```

Comment:
Maybe we can add all the layers of a scene in the constructor, or add a postCreate method, where the layers are ordered. I think most game engine's dont allow layers to be created / destroyed, while the scene is shown. We could of course still add the possiblity to disable a complete layer.

But this can be done in another PR.